packages/react-heartwood-components/src/components/Toast/components/ToastWrapper/ToastWrapper.js

```diff
@@ -12,23 +13,63 @@ type Props = {
 	handleRemove: Function
 }
 
-const ToastWrapper = (props: Props) => {
-	const { toasts, handleRemove } = props
-
-	return (
-		<div className="toasts-wrapper">
-			<VelocityTransitionGroup
-				enter={{ animation: { opacity: 1, translateX: 0 }, duration: 300 }}
-				leave={{ animation: { opacity: 0, translateX: '-4px' }, duration: 0 }}
-			>
-				{toasts.map((toast, idx) => (
-					<div key={idx} className="toast-wrapper">
-						<Toast onRemove={() => handleRemove(idx)} {...toast} />
-					</div>
-				))}
-			</VelocityTransitionGroup>
-		</div>
-	)
+type State = {
+	toasts: Array<ToastProps>,
+	timeouts: Object
 }
 
-export default ToastWrapper
+const defaultTimeout = 3000
+
+export default class ToastWrapper extends Component<Props, State> {
+	state = {
+		toasts: [],
+		timeouts: {}
+	}
+	timeouts = {}
+	static getDerivedStateFromProps(props: Props, state: State) {
+		const uniqToasts = uniqBy(props.toasts, 'id')
+		let timeouts = { ...state.timeouts }
+		uniqToasts.forEach(toast => {
+			if (toast.timeout !== 'never') {
+				timeouts[toast.id] = setTimeout(
+					() => {
+						props.handleRemove(toast.id)
+					},
+					typeof toast.timeout === 'number' ? toast.timeout : defaultTimeout
+				)
+			}
+		})
+		return { toasts: uniqToasts, timeouts }
```

Comment:
I don't see where you're accessing `state.toasts`; is that necessary, or do you mean to access the newly unique toasts in `render()`?

Reply:
Good call — they should be accessed in `render()`. We need it here to set the right timeouts.